channels/layers.py

```diff
@@ -173,13 +176,20 @@ def non_local_name(self, name):
 
 
 class InMemoryChannelLayer(BaseChannelLayer):
-    """
-    In-memory channel layer implementation for testing purposes.
-    """
+    '''
+    In-memory channel layer implementation
+    '''
+    local_poll_interval = 0.01
 
-    def __init__(self, *args, **kwargs):
-        super(InMemoryChannelLayer, self).__init__(*args, **kwargs)
+    def __init__(self, expiry=60, capacity=100, channel_capacity=None, **kwargs):
+        super().__init__(expiry=expiry, capacity=capacity, channel_capacity=channel_capacity, **kwargs)
         self.channels = {}
+        self.groups = {}
+        self.thread_lock = threading.Lock()
```

Comment:
I'm not sure a `threading.Lock` makes sense here - you shouldn't need it since you're writing async code. Did you find a reason to add it?